### PR TITLE
Raise error for internal server error

### PR DESCRIPTION
**Pull request by shioyama**
I added a method `Openlogi::Response#validate!` which raises exceptions on particular responses, and added internal server error to that. Now we can just add any other response errors we want to raise an exception on to this method.

Other responses are: success (200), unprocessable entity (422), duplicate entry (409), record not found (404). For now these all return an object with errors on that object without raising.

### CODE REVIEW

File: lib/openlogi/base_object.rb [hunk 1]
```diff
@@ -20,7 +20,7 @@ def initialize(attributes = {}, &block)
     property :error_description
 
     def valid?
-      errors.empty?
+      errors.nil? || errors.empty?
```
Comment (shioyama) [inline lib/openlogi/base_object.rb]: Snuck this in as well because `valid?` was crashing when `errors` was nil.

**Comment by degicat** [comment]
@rramsden Please review this. :bow::bow::bow::bow:

**Comment by shioyama** [comment]
@rramsden it's really minor

**Comment by rramsden** [comment]
LGTM if tests pass

**Pull request merged**